applications/plugins/ArticulatedSystemPlugin/src/ArticulatedSystemPlugin/ArticulatedSystemMapping.inl

```diff
@@ -156,8 +166,7 @@ void ArticulatedSystemMapping<TIn, TInRoot, TOut>::apply( typename Out::VecCoord
     // Copy the root position if a rigid root model is present
     if (m_fromRootModel && inroot)
     {
-        int index = (d_indexFromRoot.isSet())? d_indexFromRoot.getValue(): m_fromRootModel->getSize()-1;
-        out[0] = (*inroot)[index];
+        out[0] = (*inroot)[d_indexFromRoot.getValue()];
```

Comment:
I think it is dangerous to use directly d_indexFromRoot.getValue() as the check you made is ony at init() and nothing prevents the user to modify it afterwards with a bogus value

Reply:
If the user set `d_indexFromRoot`, it will be used in the previous version of the code. So you mean a check must be added?

Reply:
I guess you mean that I should add a member variable m_indexFromRoot, initialize it from d_indexFromRoot or the default value in the init(), and then use this variable elsewhere instead of d_indexFromRoot?

Reply:
If yes, in such case, should we set the data as  “only readable” from the GUI ?

Reply:
> I guess you mean that I should add a member variable m_indexFromRoot, initialize it from d_indexFromRoot or the default value in the init(), and then use this variable elsewhere instead of d_indexFromRoot?

Could be a solution yes, as there is no "Read-only" mechanism in Data (AFAIK)

> If yes, in such case, should we set the data as “only readable” from the GUI ?

at least in for a GUI point of view, it is a good idea ; just it wont prevent somebody to modify it through getData() in C++ or  with python3 (but if you do the m_indexFromRoot it will prevent any crash)